src/components/application_manager/src/resumption/resumption_data_processor.cc

```diff
@@ -566,74 +566,89 @@ void ResumptionDataProcessor::AddButtonsSubscriptions(
           (subscriptions_buttons[i]).asInt());
       application->SubscribeToButton(btn);
     }
+
+    ButtonSubscriptions button_subscriptions =
+        GetButtonSubscriptionsToResume(application);
+
+    MessageHelper::SendOnButtonSubscriptionNotificationsForApp(
+        application, application_manager_, button_subscriptions);
   }
 
-  MessageHelper::SendAllOnButtonSubscriptionNotificationsForApp(
-      application, application_manager_);
-}
+  ButtonSubscriptions ResumptionDataProcessor::GetButtonSubscriptionsToResume(
+      ApplicationSharedPtr application) const {
+    ButtonSubscriptions button_subscriptions =
+        application->SubscribedButtons().GetData();
+    auto it = button_subscriptions.find(mobile_apis::ButtonName::CUSTOM_BUTTON);
+    
+    if (it != button_subscriptions.end()) {
+      button_subscriptions.erase(it);
+    }
 
-void ResumptionDataProcessor::AddPluginsSubscriptions(
-    ApplicationSharedPtr application,
-    const smart_objects::SmartObject& saved_app) {
-  LOG4CXX_AUTO_TRACE(logger_);
+    return button_subscriptions;
+  }
+
+  void ResumptionDataProcessor::AddPluginsSubscriptions(
+      ApplicationSharedPtr application,
+      const smart_objects::SmartObject& saved_app) {
+    LOG4CXX_AUTO_TRACE(logger_);
 
-  for (auto& extension : application->Extensions()) {
-    extension->ProcessResumption(
-        saved_app,
-        [this](const int32_t app_id, const ResumptionRequest request) {
-          this->WaitForResponse(app_id, request);
-        });
+    for (auto& extension : application->Extensions()) {
+      extension->ProcessResumption(
+          saved_app,
+          [this](const int32_t app_id, const ResumptionRequest request) {
+            this->WaitForResponse(app_id, request);
+          });
+    }
   }
-}
 
-void ResumptionDataProcessor::DeleteSubscriptions(const int32_t app_id) {
-  LOG4CXX_AUTO_TRACE(logger_);
-  ApplicationSharedPtr application = application_manager_.application(app_id);
-  DeleteButtonsSubscriptions(application);
-  DeletePluginsSubscriptions(application);
-}
+  void ResumptionDataProcessor::DeleteSubscriptions(const int32_t app_id) {
+    LOG4CXX_AUTO_TRACE(logger_);
+    ApplicationSharedPtr application = application_manager_.application(app_id);
+    DeleteButtonsSubscriptions(application);
+    DeletePluginsSubscriptions(application);
+  }
+
+  void ResumptionDataProcessor::DeleteButtonsSubscriptions(
```

Comment:
@mked-luxoft why DeleteSubscriptions reveives `int32_t` but DeleteButtonsSubscriptions recives ApplicationSharedPtr? Please make it consistant

Reply:
@LuxoftAKutsan  fixed in 0ec2a534b5e4af4400ca173711e30b112a0910ae